api/src/app.ts

```diff
@@ -0,0 +1,193 @@
+import { config } from "dotenv";
+import { resolve } from "path";
+
+config({ path: resolve(__dirname, "..", ".env") });
+
+import express, { Request, Response, NextFunction } from "express";
+import Grant from "./models";
+
+const app = express();
+
+/**
+ * List of available, supported API versions.
+ */
+const apiVersions = [1];
+
+/**
+ * The version in the header is namespaced, specified as:
+ *   gov.usda.fs.nrm.ga.version={VERSION}
+ *
+ * As presently written, versions can only be whole integers.
+ *
+ */
+const apiVersionHeader = /gov\.usda\.fs\.nrm\.ga\.api\.version=(\d+)/;
+
+/**
+ * This matches the first segment of the request path, formatted as:
+ *   v{VERSION}
+ *
+ * As presently written, versions can only be whole integers.
+ *
+ */
+const apiVersionPath = /^v(\d+)$/;
+
+/**
+ * @todo Set using environment variables for security hardening.
+ */
+app.use(function (req, res, next) {
+  res.setHeader("Access-Control-Allow-Origin", "*");
+  res.setHeader("Access-Control-Allow-Methods", "*");
+  res.setHeader("Access-Control-Allow-Headers", "*");
+  next();
+});
+
+const awaitErrorHandlerFactory = (middleware: any) => {
+  return async (request: Request, response: Response, next: NextFunction) => {
+    try {
+      await middleware(request, response, next);
+    } catch (err) {
+      console.error(err);
+      next(err);
+    }
+  };
+};
+
+const versionMessages = {
+  200: {
+    title: "OK",
+    detail: "Everything is fine.",
+  },
+  400: {
+    title: "No API version specified",
+    detail: `An API version must be specified, but no version was specified.
+        To specify a version, use one of the following approaches:
+        (1) Specify the version in the first segment of the path, e.g. \`/v1/grants\`
+        (2) Specify the version in the Accept header: \`Accept: application/vnd.api+json; gov.usda.fs.ga.api.version=1\``,
+  },
+  404: {
+    title: "Version not found",
+    detail: `The version specified is not an available API version. Please choose from the following available versions: ${apiVersions}`,
+  },
+  409: {
+    title: "Conflicting versions requested",
+    detail:
+      "The Accept header and the request path each requested different, conflicting versions. Please ensure that only one version is specified. You can set both the path and header to the same version, or you can request a version using only the path or only the header.",
+  },
+};
+
+/**
+ * Adds a `version` property to Express.Request, so we can attach
+ * that metadata as we resolve the version.
+ */
+declare global {
```

Comment:
**suggestion:** refactor the version routing functionality into its own module, and `app.use` it here.
**suggestion:** find a way to declare this attribute/type without doing so in the `global` namespace. `global` is often a code smell.

Reply:
Added comments to this effect.